library/MedEx/API.php

```diff
@@ -0,0 +1,2092 @@
+        ?>
+      <div class="row">
+        <div class="col-sm-12">
+            <div class="showRecalls" id="show_recalls" style="text-align:center;margin:40 auto;">
+                <?php if ($logged_in) {
+                    foreach ($results['events'] as $event) {
+                        if ($event['M_group'] != 'RECALL') continue;
+                        $icon = $this->get_icon($event['M_type'],'SCHEDULED');
+                        if ($event['E_timing'] =='1') $action = "before";
+                        if ($event['E_timing'] =='2') $action = "before (PM)";
+                        if ($event['E_timing'] =='3') $action = "after";
+                        if ($event['E_timing'] =='4') $action = "after (PM)";
+                        $current_events .=  $icon." ".$event['E_fire_time']." days ".$action."<br />";
+                    }
+                ?>
+                <div class="borderShadow" style="position:absolute;top:20px;right:10px;width:200px;font-size:0.9em;">
+                    <a class="fa fw fa-plus-square" title="Add a New Recall" id="BUTTON_new_recall_menu" style="float:left;" href="<?php echo $GLOBALS['web_root']; ?>/interface/main/messages/messages.php?go=addRecall"></a>
+                    <b><u>MedEx <?php echo xlt('Recall Schedule'); ?></u></b><br />
+                    <span>
+                        <?php echo $current_events; ?>
+                    </span>
+                </div>
+                <?php } ?>
+                <div class="title"><?php echo xlt('Recall Board'); ?></div>
+                <div name="div_response" id="div_response"><?php echo xlt('Persons needing a recall, no appt scheduled yet'); ?>.</div>
+                <div name="message" id="message" class="warning"></div>
+                <ul class="nav nav-tabs <?php echo $reminder_bar; ?>">
+                    <li class="active whitish"><a href="#tab-all" data-toggle="tab"><?php echo xlt('All'); ?></a></li>
+                    <li class="whitish"><a href="#tab-pending" data-toggle="tab"><?php echo xlt('Events Scheduled'); ?></a></li>
+                    <li class="yellowish"><a href="#tab-processing" data-toggle="tab"><?php echo xlt('In-process'); ?></a></li>
+                    <li class="greenish"><a href="#tab-complete" data-toggle="tab"><?php echo xlt('Successfully Completed'); ?></a></li>
+                    <li class="reddish"><a href="#tab-manual" data-toggle="tab"><?php echo xlt('Manual Processing Required'); ?></a></li>
+                </ul>
+                <div class="tab-content">
+                    <div class="tab-pane active" id="tab-all">
+                        <?php  
+                            $this->recall_board_top($logged_in); 
+                            echo $processed['ALL'];
+                            $this->recall_board_bot($logged_in);
+                        ?>
+                    </div>
+                    <?php if ($logged_in) { ?>        
+                    <div class="tab-pane"
```

Comment:
need attr() around each of these

(Will do a line by line review of html escaping and translation stuff on next review; goal of this review are bigger picture items)